packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -302,6 +311,39 @@ export default class OperatorModeStateService extends Service {
       .map((stack) => stack[stack.length - 1]);
   }
 
+  getOpenCardIds(selectedCardRef?: ResolvedCodeRef): string[] | undefined {
+    // selectedCardRef is only needed for determining open playground card id in code submode
+    if (this.state.submode === Submodes.Code && selectedCardRef) {
+      let moduleId = internalKeyFor(selectedCardRef, undefined);
+      return [this.playgroundPanelService.getSelection(moduleId)?.cardId];
+    }
+    if (this.state.submode === Submodes.Interact) {
+      return this.topMostStackItems()
+        .filter((stackItem: StackItem) => stackItem)
+        .map((stackItem: StackItem) => stackItem.url)
+        .filter(Boolean) as string[];
+    }
+    return;
+  }
+
+  getOpenCards = restartableTask(async (selectedCardRef?: ResolvedCodeRef) => {
+    let cardIds = this.getOpenCardIds(selectedCardRef);
+    if (!cardIds) {
+      return;
+    }
+    let cards = (await Promise.all(cardIds.map((id) => this.store.peek(id))))
+      .filter(Boolean)
+      .filter(isCardInstance);
+    return cards;
```

Comment:
@habdelra is the store usage here correct in this context? I have the card ids, the existing SendAiAssistantMessage command input is expecting card defs